### PR TITLE
[Feature] Added ambient light

### PR DESCRIPTION
**Pull request by AnotherFoxGuy**
Requested by @DarthCainRor 
#374

Now you can add ambient lights to your object by adding this to your odef file

```
;pointlight x, y, z, dirx, diry, dirz, r, g, b, range
pointlight -13, 1, 6, -1, 0, -1, 10, 5, 0, 60
```
```
;spotlight x, y, z, dirx, diry, dirz, r, g, b, range, innerAngle, outerAngle 
spotlight  -13, 1, 6, -1, 0, -1, 10, 5, 0, 60,30,45
```

### CODE REVIEW

File: source/main/terrain/TerrainObjectManager.cpp [hunk 1]
```diff
@@ -1176,6 +1176,83 @@ void TerrainObjectManager::loadObject(const Ogre::String &name, const Ogre::Vect
 				continue;
 			}
 
+			if (!strncmp("spotlight", ptline, 9))
+			{
+				Vector3 lpos, ldir;
+				float lrange = 10, innerAngle = 45, outerAngle = 45;
+				ColourValue lcol;
+				String lname = "spotlight_" + TOSTRING(Math::RangeRandom(1000, 9999));
```
Comment (Hiradur) [inline source/main/terrain/TerrainObjectManager.cpp]: Does `lname` have to be unique? If so then there is now guarantee here that it's unique.

Comment (AnotherFoxGuy) [inline source/main/terrain/TerrainObjectManager.cpp]: No, it doesn't. it is only there for debugging.

**Comment by wilkgr76** [comment]
Ooooh, nice :smile: 
Sometimes I was wondering why the maps have streetlights if they don't work...

**Comment by ulteq** [comment]
Looks good for me.

**Pull request merged**